device/pcie/pci_device.cpp

```diff
@@ -39,26 +80,21 @@ static PciDeviceInfo read_device_info(int fd)
     return PciDeviceInfo{info.out.vendor_id, info.out.device_id, info.out.pci_domain, bus, dev, fn};
 }
 
-static int determine_numa_node(int fd)
-{
-    const auto device_info = read_device_info(fd);
-    const auto sysfs_path = fmt::format("/sys/bus/pci/devices/{:04x}:{:02x}:{:02x}.{}/numa_node",
-                                        device_info.pci_domain, device_info.pci_bus,
-                                        device_info.pci_device, device_info.pci_function);
-
-    std::ifstream numa_file(sysfs_path);
-    int numa_node = -1;
-    if (numa_file >> numa_node) {
-        return numa_node;
+static tt::ARCH detect_arch(uint32_t pcie_device_id, uint32_t pcie_revision_id) {
+    if (pcie_device_id == GS_PCIE_DEVICE_ID){
+        return tt::ARCH::GRAYSKULL;
+    } else if (pcie_device_id == WH_PCIE_DEVICE_ID && pcie_revision_id == 0x01){
+        return tt::ARCH::WORMHOLE_B0;
+    } else if (pcie_device_id == WH_PCIE_DEVICE_ID){
+        // TODO: did we ship any of these?  I've never seen one.  Can we stop
```

Comment:
I left similar comment to one of @broskoTT PRs. I think deprecating Wormhole is totally fine, I think we don't have a system in the company that is running anything on these cards

Reply:
I'll leave it alone for now, since it's unrelated to this PR.  Created https://github.com/tenstorrent/tt-umd/issues/158, I will remove the comment.